Dockerfile

```diff
@@ -226,4 +226,5 @@ COPY --from=build /opt/warehouse/ /opt/warehouse/
 COPY . /opt/warehouse/src/
 
 # We cannot run `postgres` as root, so add a user to run the application
+RUN chown nobody /opt/warehouse/src
```

Comment:
this _shouldn't_ be necessary and if it is it indicates that we are not mounting something that should be mounted.

Reply:
Coverage is trying to create the `/opt/warehouse/src/dev/.coverage` file during the test run as `nobody`, but since the directory is owned by the user that checked out the directory on the host, it fails. AFAICT, there's no other way to change permissions/ownership of a mounted directory.

Reply:
after `make purge` and `make tests` on `main`:

```
pancack:warehouse ee$ docker compose exec web ls -alhtr /opt/warehouse/src/dev/
total 74M
-rw-r--r-- 1 nobody nogroup  68M Jan 16 18:41 example.sql.xz
drwxr-xr-x 3 nobody nogroup   96 Jan 16 18:41 flake8
-rw-r--r-- 1 nobody nogroup 1.3K Jan 16 18:41 notdatadog.py
-rw-r--r-- 1 nobody nogroup 3.9K Jan 31 18:19 environment
drwxr-xr-x 6 nobody nogroup  192 Feb 20 18:49 .pytest_cache
drwxr-xr-x 8 nobody nogroup  256 Feb 20 18:49 .
-rw-r--r-- 1 nobody nogroup 5.9M Feb 20 18:50 .coverage
drwxr-xr-x 1 root   root      14 Feb 20 18:52 ..
pancack:warehouse ee$ ls -alhtr dev/
total 150216
-rw-r--r--   1 ee  staff    67M Jan 16 13:41 example.sql.xz
drwxr-xr-x   3 ee  staff    96B Jan 16 13:41 flake8
-rw-r--r--   1 ee  staff   1.2K Jan 16 13:41 notdatadog.py
-rw-r--r--   1 ee  staff   3.9K Jan 31 13:19 environment
drwxr-xr-x  38 ee  staff   1.2K Feb 20 13:49 ..
drwxr-xr-x   6 ee  staff   192B Feb 20 13:49 .pytest_cache
drwxr-xr-x   8 ee  staff   256B Feb 20 13:49 .
-rw-r--r--   1 ee  staff   5.8M Feb 20 13:50 .coverage
```

Reply:
At least for me:

```
$ docker compose exec web ls -alhtr /opt/warehouse/src/dev/
total 68M
drwxr-xr-x 2 root   root    4.0K Oct  5  2022 tufkeys
-rw-r--r-- 1   1001    1002  68M Feb 23  2023 example.sql.xz
-rw-r--r-- 1   1001    1002 1.3K Aug 18  2023 notdatadog.py
drwxr-xr-x 3   1001    1002 4.0K Jan  3 19:25 flake8
-rw-r--r-- 1   1001    1002 3.9K Feb 12 17:32 environment
drwxr-xr-x 3 nobody nogroup 4.0K Feb 20 17:36 .pytest_cache
drwxr-xr-x 2 nobody nogroup  12K Feb 20 17:43 htmlcov
drwxr-xr-x 6   1001    1002 4.0K Feb 20 17:49 .
-rw-r--r-- 1 nobody nogroup 512K Feb 20 17:49 .coverage
drwxr-xr-x 1 root   root    4.0K Feb 20 19:01 ..
```

Reply:
curiosity:

```
pancack:warehouse ee$ docker compose version
Docker Compose version v2.23.3
pancack:warehouse ee$ docker version
Client:
 Version:           25.0.3
 API version:       1.44
 Go version:        go1.21.6
 Git commit:        4debf41
 Built:             Tue Feb  6 21:13:26 2024
 OS/Arch:           darwin/arm64
 Context:           orbstack

Server: Docker Engine - Community
 Engine:
  Version:          25.0.3
  API version:      1.44 (minimum version 1.24)
  Go version:       go1.21.6
  Git commit:       f417435
  Built:            Tue Feb  6 21:14:35 2024
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          v1.7.13
  GitCommit:        7c3aca7a610df76212171d200ca3811ff6096eb8
 runc:
  Version:          1.1.12
  GitCommit:        51d5e94601ceffbbd85688df1c928ecccbfa4685
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
  ```